src/ts/plugins/server/GitlabHostingConnector.ts

```diff
@@ -146,12 +146,32 @@ export default class GitlabHostingConnector extends GitlabConnector implements H
     return `${projectUrl}/pages`
   }
 
-  async getGitlabJobLogsUrl(session: GitlabSession, websiteId: WebsiteId, projectUrl: string): Promise<string> {
-    const jobs = await this.callApi(session, `api/v4/projects/${websiteId}/jobs`, 'GET')
-    return `${projectUrl}/-/jobs/${jobs[0].id}`
+  async getGitlabJobLogsUrl(session: GitlabSession, websiteId: WebsiteId, job: PublicationJobData, { startJob, jobSuccess, jobError }: JobManager, projectUrl: string, tag): Promise<string> {
+    let jobs = await this.callApi(session, `api/v4/projects/${websiteId}/jobs`, 'GET')
+    // waiting for the job corresponding to the current tag
+    let i = 0
+    setTimeout (() => {
+      while (jobs[0].ref !== tag && i<20) {
+        jobs = this.callApi(session, `api/v4/projects/${websiteId}/jobs`, 'GET')
```

Comment:
Although there's a condition to limit the loop to 20 iterations, the loop does not include any delay or await between iterations. It will make many calls to callApi for each timeout

Reply:
I don't think so, the setTimeout is tune with a delay and I have mesured the mean number f call that are between 0 and 5 cycles. What I am missing ?

Reply:
you don't wait for an answer (await)
so jobs will be a promise, not the actual result
and it will run 100 times without waiting for a result

Reply:
just to verify my comprehension, this should work better ? 
`    setTimeout (async () => {
      while (jobs[0].ref !== tag && i<20) {
        jobs = await this.callApi(session, `api/v4/projects/${websiteId}/jobs`, 'GET')
        i++
      }
    }, 100)`

Reply:
no because the setTimeout will not wait that the async finishes, you have multiple calls in parallel

check my proposal of algorithm, this will do a call, wait for the response, wait a little, loop

Reply:
> check my proposal of algorithm, this will do a call, wait for the response, wait a little, loop

your algo will do: wait a little, start a call but don't get the result and don't wait, loop

Reply:
sorry for my lack of skills in async js field. I will try again !